docs/src/developing/backwards-compatibility.md

```diff
@@ -40,6 +41,10 @@ updates of a particular `MINOR` version release.
 deprecated features. Client SDKs and tooling will begin using new features and endpoints
 that were enabled in the previous `MAJOR` version.
 
+The Rust SDK will not have any future `MAJOR` releases, and will stay on `1.X` to
+provide a better developer experience. More details in
+[Why not just use semver](#why-not-just-use-semver).
```

Comment:
I wonder if this is a bit too restrictive.
There seems to be no reason to say we will never increase the major release number.
It does seem quite unlikely at the moment.
But things may change.

We do want to be explicit about the currently chosen default policy of staying on `1.x`.
So we can, maybe, say something along the lines of:

> We do not expect `MAJOR` version to be changed.  Any breaking changes will be done though the deprecation process described in [...] in a `MINOR` version.

If you think it is too subtle, it is probably fine to keep the more assertive "no `MAJOR` version updates" statement as well.

Also, this paragraph seems to contradict the previous one, that explain when happens in a major version update.  If we are saying they never happen, there is no point in this explanation.

Reply:
> I wonder if this is a bit too restrictive.

You're right, I think I was a little too inspired by our chats.

> Also, this paragraph seems to contradict the previous one, that explain when happens in a major version update. If we are saying they never happen, there is no point in this explanation.

And you're right about the contradiction -- this document also covers RPC endpoints, and I didn't want to have the policy apply to both RPC and SDK.

I'll make it clearer which parts cover RPC and which parts cover the SDK.